File: app_mobile/kmp/androidApp/src/main/java/com/ansmi/gestsquadre/kmp/data/TocMessageStorage.kt
package com.ansmi.gestsquadre.kmp.data

import android.content.Context
import com.ansmi.gestsquadre.shared.model.formatTocPanelMessage

/**
 * Ultimo messaggio TOC da mostrare nel pannello blu (sopravvive a chiusura app / processo kill).
 */
class TocMessageStorage(
    context: Context,
) {
    private val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)

    fun save(message: String) {
        val trimmed = message.trim()
        if (trimmed.isEmpty()) {
            clear()
            return
        }
        prefs.edit().putString(KEY, trimmed).apply()
    }

    fun load(): String? {
        val raw = prefs.getString(KEY, null)?.trim().orEmpty()
        return raw.ifEmpty { null }
    }

    fun clear() {
        prefs.edit().remove(KEY).apply()
    }

    companion object {
        const val PREFS_NAME = "gest_squadre_prefs"
        private const val KEY = "last_toc_message"

        fun formatDisplayMessage(
            title: String,
            body: String,
        ): String = formatTocPanelMessage(title, body).orEmpty()
    }
}

File: app_mobile/kmp/iosApp/iosApp/TocMessageStorage.swift
import Foundation

/// Ultimo messaggio TOC nel pannello blu (sopravvive a chiusura app).
final class TocMessageStorage {
    static let shared = TocMessageStorage()

    private let defaults = UserDefaults.standard
    private let key = "last_toc_message"

    private init() {}

    func save(message: String) {
        let trimmed = message.trimmingCharacters(in: .whitespacesAndNewlines)
        if trimmed.isEmpty {
            clear()
            return
        }
        defaults.set(trimmed, forKey: key)
    }

    func load() -> String? {
        let raw = defaults.string(forKey: key)?.trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
        return raw.isEmpty ? nil : raw
    }

    func clear() {
        defaults.removeObject(forKey: key)
    }

    static func formatDisplayMessage(title: String, body: String) -> String? {
        let t = title.trimmingCharacters(in: .whitespacesAndNewlines)
        let b = body.trimmingCharacters(in: .whitespacesAndNewlines)
        if t.isEmpty && b.isEmpty { return nil }
        return b.isEmpty ? t : "\(t): \(b)"
    }
}
